File: app/src/main/java/com/example/cookbook/database/dao/IngredientDao.java
package com.example.cookbook.database.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.example.cookbook.models.Ingredient;

@Dao
public interface IngredientDao {

    @Query("SELECT * FROM Ingredient WHERE ingredientId = :ingredientId")
    LiveData<Ingredient> getIngredient(long ingredientId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertIngredient(Ingredient ingredient);
}

File: app/src/main/java/com/example/cookbook/database/dao/IngredientDao.kt
package com.example.cookbook.database.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.example.cookbook.models.IngredientData
import com.example.cookbook.models.IngredientDatabase

@Dao
interface IngredientDao {
    @Query("SELECT * FROM IngredientDatabase WHERE ingredientDatabaseId= :id")
    suspend fun getIngredientDatabase(id:Long):IngredientDatabase

    @Query("SELECT * FROM IngredientDatabase")
    suspend fun getIngredientList():List<IngredientDatabase>

    @Update
    suspend fun updateIngredient(vararg ingredients:IngredientData)

    @Delete
    suspend fun deleteIngredient(vararg ingredients:IngredientData)

    @Insert
    suspend fun addIngredient(ingredient:IngredientData):Long
}